python/servo/visual_studio.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2024 The Servo Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+import json
+import os
+import subprocess
+import sys
+from typing import List
+
+import servo.platform
+
+
+def find_highest_msvc_version_ext():
+    def vswhere(args):
+        program_files = (os.environ.get('PROGRAMFILES(X86)')
+                         or os.environ.get('PROGRAMFILES'))
+        if not program_files:
+            return []
+        vswhere = os.path.join(program_files, 'Microsoft Visual Studio', 'Installer', 'vswhere.exe')
+        if not os.path.exists(vswhere):
+            return []
+        output = subprocess.check_output([vswhere, '-format', 'json'] + args).decode(errors='ignore')
+        return json.loads(output)
+
+    for install in vswhere(['-products', '*',
+                            '-requires', 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64',
+                            '-requires', 'Microsoft.VisualStudio.Component.Windows10SDK']):
+        version = install['installationVersion'].split('.')[0] + '.0'
+        yield (install['installationPath'], version)
+
+
+def find_highest_msvc_version():
+    editions = ["Enterprise", "Professional", "Community", "BuildTools"]
+    prog_files = os.environ.get("ProgramFiles(x86)")
+    base_vs_path = os.path.join(prog_files, "Microsoft Visual Studio")
+
+    # TODO(mrobinson): Add support for Visual Studio 2022.
+    vs_versions = ["2019"]
+    versions = {
+        ("2019", "vs"): "16.0",
```

Comment:
Is the second component needed in the key tuple? It seems be hardcoded in line 51?

Reply:
I don't think this is necessary. This was probably left over from some dead code. I'm happy to remove it.

Reply:
I've removed this as well as cleaned up a bit some of the surrounding code, returning a `dataclass` from the caller, which should help ensure type safety.